### PR TITLE
CraneLoadChartLegendData | multiple sets of legends

### PR DESCRIPTION
**Pull request by Minyewoo**

### CODE REVIEW

**Review by a-givertzman — COMMENTED** [review]
my suggestion

File: test/unit/charts/crane_load_chart/swl_data_converter/swl_data_converter_ordering_test.dart [hunk 1]
```diff
@@ -23,11 +24,11 @@ void main() {
         height: height, 
         rawWidth: width, 
         width: width,
-        legendData: CraneLoadChartLegendData(
-          limits: limitSet,
-          colors: colorSet,
-          names: nameSet,
-        ),
+        legendData: await CraneLoadChartLegendJson(
```
[the remark this answers is not in the document]
Comment (Minyewoo) [reply test/unit/charts/crane_load_chart/swl_data_converter/swl_data_converter_ordering_test.dart]: Done

File: lib/src/charts/crane_load_chart/crane_load_chart.dart [hunk 2]
```diff
@@ -92,15 +93,15 @@ class _CraneLoadChartState extends State<CraneLoadChart> {
   @override
   void initState() {
     _showGrid = widget._showGrid;
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart.dart:95]: Let`s pass _showGrid through constructor

File: lib/src/charts/crane_load_chart/crane_load_chart.dart [hunk 3]
```diff
@@ -92,15 +93,15 @@ class _CraneLoadChartState extends State<CraneLoadChart> {
...
-    log(_debug, '[_CraneLoadChartState.initState] legendData limits: ', _swlDataCache.legendData.limits);
-    log(_debug, '[_CraneLoadChartState.initState] legendData colors: ', _swlDataCache.legendData.colors);
-    log(_debug, '[_CraneLoadChartState.initState] legendData names: ', _swlDataCache.legendData.names);
+    _log.info('[_CraneLoadChartState.initState] legendData limits: ', _swlDataCache.legendData.limits);
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart.dart:96]: may be log.debug will be more suitable in this case

File: lib/src/charts/crane_load_chart/crane_load_chart.dart [hunk 4]
```diff
@@ -92,15 +93,15 @@ class _CraneLoadChartState extends State<CraneLoadChart> {
...
+    _log.info('[_CraneLoadChartState.initState] legendData colors: ', _swlDataCache.legendData.colors);
+    _log.info('[_CraneLoadChartState.initState] legendData names: ', _swlDataCache.legendData.names);
 
     final swlIndexStream = _swlIndexStream;
     if (swlIndexStream != null) {
       _swlIndexStreamSubscription = swlIndexStream.listen((event) {
-        log(_debug, '_CraneLoadChartState.swlIndexStream.listen] event: ', event);
-        log(_debug, '_CraneLoadChartState.swlIndexStream.listen] event.status: ', event.status);
+        _log.info('_CraneLoadChartState.swlIndexStream.listen] event: ', event);
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart.dart:103]: may be log.debug will be more suitable in this case

File: lib/src/charts/crane_load_chart/swl_data_converter.dart [hunk 5]
```diff
@@ -61,20 +61,27 @@ class SwlDataConverter implements CraneLoadChartData {
...
+    ];
   }  
   ///
-  Color _pickSwlColor(double swl, List<_Gap> segments) {
+  Future<Color> _pickSwlColor(int layerIndex, double swl, List<_Gap> segments) async {
     final colorIndex = segments.indexWhere(
       (segment) => segment.contains(swl),
     );
-    return _legendData.colors.elementAt(colorIndex);
+    final colors = await _legendData.colors;
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/swl_data_converter.dart:83]: pass colors as input parameter

**Review by a-givertzman — COMMENTED** [review]
first step

File: lib/src/charts/crane_load_chart/swl_data_converter.dart (hunk 5 above)
Comment (Minyewoo) [reply lib/src/charts/crane_load_chart/swl_data_converter.dart]: Done

File: lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            .elementAt(_swlIndex);
+          final names = (snapshot.data![1] as Iterable<Iterable<String>>)
+            .elementAt(_swlIndex);
+          return SizedBox(
+            width: _computeWidth(
+              DefaultTextStyle.of(context).style.fontSize ?? 100, 
+              names, 
+              _width,
+            ),
+            child: ListView.separated(
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart:45]: Replace ListView with Column widget

File: lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                  textAlign: _textAlign,
+                ),
+              ), 
+              separatorBuilder: (_, __) => SizedBox(height: padding), 
+              itemCount: names.length,
+            ),
+          );
+        }
+        else {
+          return Padding(
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart:64]: will be enough:
```
          return const Center(
            child: CircularProgressIndicator(),
          );          
```

File: lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import 'package:hmi_widgets/src/charts/crane_load_chart/swl_data_cache.dart';
+import 'package:hmi_core/hmi_core.dart';
+///
+class CraneLoadChartLegendWidget extends StatelessWidget {
+  final SwlDataCache _swlDataCache;
+  final double? _width;
+  final TextAlign? _textAlign;
+  final int _swlIndex;
+  ///
+  const CraneLoadChartLegendWidget({
```
Comment (a-givertzman) [inline lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart:11]: add constrictor parameters with default margin, padding = Setting('padding', factor: 0.5)

**Review by a-givertzman — COMMENTED** [review]
let't improve CraneLoadChartLegendWidget after AppSettings will be merged

**Review by a-givertzman — APPROVED** [review]

File: lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart (hunk 6 above)
Comment (a-givertzman) [reply lib/src/charts/crane_load_chart/crane_load_chart_legend_widget.dart]: keepped ListView - as more suitable in current case

**Pull request merged**